From 0a35d917be639f03cb81f2e36d716524ba481596 Mon Sep 17 00:00:00 2001
From: Vinay Kumar Pabboju <vinayx.kumar.pabboju@intel.com>
Date: Mon, 19 Dec 2022 11:23:17 +0000
Subject: [PATCH] Fix for IFWI Version showing in Running Image

Enabled Bios related code which is responsible for showing
IFWI Firmware version in Running image

Tested:
Confirmed that IFWI Firmware Version is showing in Running image.

Signed-off-by: Vinay Kumar Pabboju <vinayx.kumar.pabboju@intel.com>
Upstream-Status: Pending
---
 redfish-core/include/redfish.hpp | 2 +-
 redfish-core/lib/bios.hpp        | 6 ++++--
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/redfish-core/include/redfish.hpp b/redfish-core/include/redfish.hpp
index f66b9d6d..fc88fa1a 100644
--- a/redfish-core/include/redfish.hpp
+++ b/redfish-core/include/redfish.hpp
@@ -225,8 +225,8 @@ class RedfishService
 
         requestRoutesSystems(app);
 
-#ifndef BMCWEB_CTL_DISABLE_ROUTES
         requestRoutesBiosService(app);
+#ifndef BMCWEB_CTL_DISABLE_ROUTES
         requestRoutesBiosReset(app);
         requestRoutesBiosSettings(app);
         requestRoutesBiosAttributeRegistry(app);
diff --git a/redfish-core/lib/bios.hpp b/redfish-core/lib/bios.hpp
index a7869305..9d5f2d84 100644
--- a/redfish-core/lib/bios.hpp
+++ b/redfish-core/lib/bios.hpp
@@ -157,6 +157,7 @@ inline void
                                    systemName);
         return;
     }
+#ifndef BMCWEB_CTL_DISABLE_ROUTES
     asyncResp->res.jsonValue["@odata.id"] = "/redfish/v1/Systems/system/Bios";
     asyncResp->res.jsonValue["@odata.type"] = "#Bios.v1_1_0.Bios";
     asyncResp->res.jsonValue["Name"] = "BIOS Configuration";
@@ -168,11 +169,11 @@ inline void
     asyncResp->res.jsonValue["Actions"]["#Bios.ChangePassword"] = {
         {"target", "/redfish/v1/Systems/system/Bios/Actions/"
                    "Bios.ChangePassword"}};
-
+#endif
     // Get the ActiveSoftwareImage and SoftwareImages
     sw_util::populateSoftwareInformation(asyncResp, sw_util::biosPurpose, "",
                                          true);
-
+#ifndef BMCWEB_CTL_DISABLE_ROUTES
     asyncResp->res.jsonValue["@Redfish.Settings"] = {
         {"@odata.type", "#Settings.v1_3_0.Settings"},
         {"SettingsObject",
@@ -255,6 +256,7 @@ inline void
         "xyz.openbmc_project.ObjectMapper", "GetObject",
         "/xyz/openbmc_project/bios_config/manager",
         std::array<const char*, 0>());
+#endif
 }
 
 inline void requestRoutesBiosService(App& app)
-- 
2.25.1

